### DISCUSSION TITLE
stat/RESULT flood on mqtt on setting vars on complex rules

### DISCUSSION THREAD

**Discussion by chaveiro** [arendst/Tasmota#11941] [Q&A] [answered]
I've setup some rules that sets many vars for each Energy parameter to later send them on a single websend command like bellow.
Rule executes every 5 seconds.
Every time a variable is set a stat/RESULT message is published to mqtt.
Enven the timer expiring, websend done  and TuyaSend done send additional stat/RESULT messages on mqtt.

Are this messages useful to anything ? 
Could the mqtt message be silent with any option? 

Sample rules bellow:

```
Rule1 
  ON System#Init DO backlog PowerOnState 4; tuyasend1 1,1; RuleTimer1 5; ENDON 
  ON Rules#Timer=1 DO backlog TuyaSend8; RuleTimer1 5 ENDON

rule2
ON energy#power DO Var3 %value% ENDON
ON energy#voltage DO Var1 %value% ENDON
ON energy#total DO Var4 %value% ENDON
ON energy#apparentpower DO Var5 %value% ENDON
ON energy#reactivepower DO Var6 %value% ENDON
ON energy#Factor DO Var7 %value% ENDON
ON energy#Today DO Var8 %value% ENDON
ON energy#Yesterday DO Var9 %value% ENDON
ON energy#current DO Var2 %value% ENDON
ON Wifi#Connected DO backlog buzzer 2,1,1; RuleTimer2 5 ENDON 
ON Wifi#Disconnected DO backlog buzzer 1,3,1; RuleTimer2 0 ENDON 
ON Rules#Timer=2 DO backlog event SENDEMON; RuleTimer2 5 ENDON
ON event#SENDEMON DO WEBSEND [emon.server.com:80] /input/post.json?node=PowemeterLago&json={P:%Var3%,I:%Var2%,V:%Var1%,S:%Var5%,Q:%Var6%,Pf:%Var7%,Total:%Var4%,Today:%Var8%,Yesterday:%Var9%}&devicekey=SOMEKEY ENDON


```

**Comment by sfromis** [comment, marked as answer]
No such options. The general idea is that a little bit of network traffic has no real cost, and you should just not look at them if you have no use for these detais. You can make filtering via MQTT subscription a bit easier using [SetOption4](https://tasmota.github.io/docs/Commands/#setoption4) to get differentiated topics instead of /RESULT.

Some logic internally in Tasmota (like for rule triggers) are based on the JSON data format (result) from one command being available for automations.

For those energy monitoring triggers, I'd probably want to use `tele-Energy#Power` and so on to only get them processed each TelePeriod, instead of all the time, possibly combined with PowerDelta to get extra reporting when consumption changes.
